lib/screens/v2/widget_instance/line_map.ex

```diff
@@ -52,29 +81,98 @@ defmodule Screens.V2.WidgetInstance.LineMap do
     def audio_view(_instance), do: ScreensWeb.Views.V2.Audio.LineMapView
   end
 
-  def serialize_stops(current_stop, stops) do
+  def serialize_stops(current_stop, stops, reverse_stops, is_terminal?) do
     current_stop_index = Enum.find_index(stops, fn %{id: stop} -> stop == current_stop end)
 
-    stops
-    |> Enum.slice(0, current_stop_index + 3)
-    |> Enum.with_index()
-    |> Enum.map(fn {%Screens.Stops.Stop{name: label}, i} ->
-      %{
-        label: label,
-        downstream: i > current_stop_index,
-        current: i == current_stop_index,
-        terminal: i == 0 or i == length(stops) - 1
-      }
-    end)
-    |> Enum.reverse()
+    forward_stops =
+      stops
+      |> Enum.slice(0, current_stop_index + @num_future_stops + 1)
+      |> Enum.with_index()
+      |> Enum.map(fn {%Screens.Stops.Stop{name: label}, i} ->
+        %{
+          label: label,
+          downstream: i > current_stop_index,
+          current: i == current_stop_index,
+          terminal: i == 0 or i == length(stops) - 1
+        }
+      end)
+      |> Enum.reverse()
+
+    if is_terminal? do
+      backward_stops =
+        reverse_stops
+        |> Enum.reverse()
+        |> Enum.drop(1)
+        |> Enum.with_index()
+        |> Enum.map(fn {%Screens.Stops.Stop{name: label}, i} ->
+          %{
+            label: label,
+            downstream: false,
+            current: false,
+            terminal: i == length(reverse_stops) - 2
```

Comment:
Could you explain why it's `length(reverse_stops) - *2*`? (Also would this ever happen? I think all of our lines are long enough that you wouldn't see the whole length from one terminus to the other on the e-ink line map)